### PR TITLE
Add ruff lint workflow and address linting issues.

### PR DESCRIPTION
**Pull request by neelam-kushwah**
#36 , #14 

Fix linting issues and refactor code for PEP 8 compliance ( except for the max length limit check, The PEP 8 guideline suggests a maximum line length of 79 characters for Python code, However, in practice, many projects and developers find this limit too restrictive for modern development environments. With wider screens and IDEs commonly setting default limits of 120 characters, it's common to see larger line length limits used in practice, typically around 100 or 120 characters. By allowing longer lines, code can often be more readable and maintainable, especially for function calls, method chains, or longer strings)

- Organized imports in the correct order.
- Refactored function and variable names to follow snake_case naming convention.
- Ensured class names start with a capital letter.
- Removed commented-out code to clean up the codebase.
- Applied PEP 8 formatting rules to ensure consistency.
- Addressed Pyflakes rules to fix any undefined names and other issues.

### CODE REVIEW

File: uprotocol/transport/validate/uattributesvalidator.py [hunk 1]
```diff
@@ -108,7 +104,7 @@ def is_expired(u_attributes: UAttributes) -> bool:
         The message has expired when the current time is greater
         than the original UUID time
         plus the ttl attribute.
-        @param uAttributes UAttributes with time to live value.
+        @param u_attributes UAttributes with time to live value.
```
Comment (matthewd0123) [inline uprotocol/transport/validate/uattributesvalidator.py:107]: It looks like there's a bit of inconsistency between using things like "u_attributes" and "uattributes", "u_entity" vs "uentity". Can we ensure that we choose one version and that it's consistent?

File: uprotocol/uri/factory/uentity_factory.py
Comment (matthewd0123) [inline uprotocol/uri/factory/uentity_factory.py:1]: Some files are using an "_" between the object and the descriptor. So like this file is uentity_factory, but other files are of the form uattributesbuilder. Can we choose one format and align all the files to it?

File: uprotocol/transport/validate/uattributesvalidator.py (hunk 1 above)
Comment (neelam-kushwah) [reply uprotocol/transport/validate/uattributesvalidator.py:107]: I believe it's acceptable for developers to name variables as they see fit, provided they adhere to the guidelines

File: uprotocol/uri/factory/uentity_factory.py
Comment (neelam-kushwah) [reply uprotocol/uri/factory/uentity_factory.py:1]: Same as above

File: uprotocol/transport/validate/uattributesvalidator.py (hunk 1 above)
Comment (matthewd0123) [reply uprotocol/transport/validate/uattributesvalidator.py:107]: for variable names this is fine

File: uprotocol/uri/factory/uentity_factory.py
Comment (matthewd0123) [reply uprotocol/uri/factory/uentity_factory.py:1]: for filenames, there is a lot of visibility... so a little bit of consistency would go a long way, looks a bit messy

**Review by matthewd0123 — APPROVED** [review]

**Pull request merged**